.github/workflows/bcny-cmake.yml

```diff
@@ -7,43 +7,86 @@ jobs:
   build:
     runs-on: swift-build-windows-latest-32-cores
 
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - arch: 'amd64'
+            cmake_arch: 'AMD64'
+            cpack_arch: 'x64'
+          - arch: 'arm64'
+            cmake_arch: 'ARM64'
+            cpack_arch: 'arm64'
+
     steps:
     - uses: compnerd/gha-setup-vsdevenv@main
+      with:
+        host_arch: amd64
+        components: 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64;Microsoft.VisualStudio.Component.VC.Tools.ARM64'
+        arch: ${{ matrix.arch }}
     - uses: actions/checkout@v4
       with:
         path: ${{ github.workspace }}/SourceCache/cmake
-    - run: cmake -B ${{ github.workspace }}/BinaryCache/cmake -G Ninja -S ${{ github.workspace }}/SourceCache/cmake -D CMAKE_BUILD_TYPE=Release
+    - run: >
+        cmake -B ${{ github.workspace }}/BinaryCache/cmake `
+              -G Ninja `
+              -S ${{ github.workspace }}/SourceCache/cmake `
+              -D CMAKE_BUILD_TYPE=Release `
+              -D CMAKE_SYSTEM_NAME=Windows `
+              -D CMAKE_SYSTEM_PROCESSOR=${{ matrix.cmake_arch }}
     - run: cmake --build ${{ github.workspace }}/BinaryCache/cmake
     - run: |
         cd ${{ github.workspace }}/BinaryCache/cmake
-        cpack -G WIX
-    - uses: actions/upload-artifact@v3
+        cpack -G WIX -D CPACK_SYSTEM_NAME=windows -D CPACK_TOPLEVEL_TAG=windows-${{ matrix.cpack_arch }}
+    - uses: actions/upload-artifact@v4
       with:
-        name: cmake-windows-amd64
+        name: cmake-windows-${{ matrix.arch }}
         path: ${{ github.workspace }}/BinaryCache/cmake/*.msi
+
+  release:
+    runs-on: swift-build-windows-latest-32-cores
```

Comment:
We can probably do this step on ubuntu-latest - we don't need the heavy weight machines for the release bit.

Reply:
Done